### ISSUE TITLE
yield as the default value in arrow expression should not be a yield expression

### ISSUE THREAD

**Issue by ariya** [eslint/espree#165]
Test case:

```js
function() {
    var espree = require('espree');
    var options = { ecmaFeatures: { generators: true, arrowFunctions: true, defaultParams: true } };
    var ast = espree.parse('function *g() { (x = yield) => {} }', options);
    return ast.body[0].body.body[0].expression.params[0].right.type;
}
```

Actual: YieldIdentifier.

Expected: Identifier.

**Comment by nzakas** [comment]
Thanks!

**Comment by nzakas** [comment]
Working on this.

**Comment by nzakas** [comment]
Is this true for function expressions as well as arrow functions?
